datafusion/expr/src/type_coercion/functions.rs

```diff
@@ -573,6 +573,8 @@ fn coerced_from<'a>(
         (Interval(_), _) if matches!(type_from, Utf8 | LargeUtf8) => {
             Some(type_into.clone())
         }
+        // We can go into a Utf8View from a Utf8 or LargeUtf8
+        (Utf8View, _) if matches!(type_from, Utf8 | LargeUtf8) => Some(type_into.clone()),
```

Comment:
The arrow cast kernel supports this conversion

Though I am not sure if we should allow the automatic coercion as it could potentially fail if the strings were over 2GB

Reply:
I see, the semantics is < 32 bit length `LargeUtf8` conversion is supported, otherwise, some runtime error will occur. This makes sense.